src/Umbraco.Web.UI.Client/src/common/services/clipboard.service.js

```diff
@@ -1,203 +1,276 @@
+    * @param {object} entry A object containing the properties to be saved, this could be the object of a ElementType, ContentNode, ...
+    * @param {string} displayLabel (optional) A string swetting the label to display when showing paste entries.
+    *
+    * @description
+    * Saves a single JS-object with a type and alias to the clipboard.
+    */
+    service.copy = function(type, alias, data, displayLabel) {
+        
+        var storage = retriveStorage();
+
+        var uniqueKey = data.key || data.$$hashKey || console.error("missing unique key for this content");
+        
+        // remove previous copies of this entry:
+        storage.entries = storage.entries.filter(
+            (entry) => {
+                return entry.unique !== uniqueKey;
+            }
+        );
+        
+        var entry = {unique:uniqueKey, type:type, alias:alias, data:prepareEntryForStorage(data), label:displayLabel || data.name, icon:iconHelper.convertFromLegacyIcon(data.icon)};
+        storage.entries.push(entry);
+        
+        if (saveStorage(storage) === true) {
+            notificationsService.success("Clipboard", "Copied to clipboard.");
+        } else {
+            notificationsService.success("Clipboard", "Couldnt copy this data to clipboard.");
+        }
+        
+    };
+
+
+    /**
+    * @ngdoc method
+    * @name umbraco.services.clipboardService#copyArray
+    * @methodOf umbraco.services.clipboardService
+    *
+    * @param {string} type A string defining the type of data to storing, example: 'elementTypeArray', 'contentNodeArray'
+    * @param {string} aliases An array of strings defining the alias of the data to store, example: ['banana', 'apple']
+    * @param {object} datas An array of objects containing the properties to be saved, example: [ElementType, ElementType, ...]
+    * @param {string} displayLabel A string setting the label to display when showing paste entries.
+    * @param {string} displayIcon A string setting the icon to display when showing paste entries.
+    * @param {string} uniqueKey A string prodiving an identifier for this entry, existing entries with this key will be removed to ensure that you only have the latest copy of this data.
+    *
+    * @description
+    * Saves a single JS-object with a type and alias to the clipboard.
+    */
+    service.copyArray = function(type, aliases, datas, displayLabel, displayIcon, uniqueKey) {
+        
+        var storage = retriveStorage();
+        
+        // Clean up each entry
+        var copiedDatas = datas.map(data => prepareEntryForStorage(data));
+        
+        // remove previous copies of this entry:
+        storage.entries = storage.entries.filter(
+            (entry) => {
+                return entry.unique !== uniqueKey;
+            }
+        );
+        
+        var entry = {unique:uniqueKey, type:type, aliases:aliases, data:copiedDatas, label:displayLabel, icon:displayIcon};
+
+        console.log(entry);
```

Comment:
Do we still need this console.log @nielslyngsoe ?

Reply:
No, thats a mistake